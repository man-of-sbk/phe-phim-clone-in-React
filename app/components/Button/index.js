/**
 *
 * CustomButton
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Button as AntButton, Icon } from 'antd';

const StyledButton = styled(AntButton)`
  border-color: ${props => props.borderColor};
  color: ${props => props.color};
  display: flex;
  align-items: center;

  .anticon {
    margin-right: 10px;
  }

  a {
    color: ${props => props.color}!important;
  }
`;

function Button({ children, color, iconType, className, borderColor }) {
  return (
    <StyledButton
      ghost
      color={color}
      borderColor={borderColor}
      className={className}
    >
      {iconType !== undefined && <Icon type={iconType} />}
      {children}
    </StyledButton>
  );
}

Button.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
  iconType: PropTypes.string,
  className: PropTypes.string,
};

Button.propTypes = {};

export default Button;
