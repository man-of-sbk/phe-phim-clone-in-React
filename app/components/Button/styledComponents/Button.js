/* eslint-disable prettier/prettier */
import React from 'react';
import styled from 'styled-components';
import { Button as AntButton } from 'antd';

// *** prevent styled-component pass props to generated DOM ele.
// ***** READ MORE: => https://stackoverflow.com/questions/49834251/how-to-extend-styled-component-without-passing-props-to-underlying-dom-element
const Button = styled(({ borderColor, color, ...rest }) => (
  <AntButton {...rest} />
))`
  border-color: ${props => props.borderColor};
  color: ${props => props.color};
  display: flex;
  align-items: center;

  .anticon {
    margin-right: 10px;
  }

  a {
    color: ${props => props.color}!important;
  }
`;

export default Button;
