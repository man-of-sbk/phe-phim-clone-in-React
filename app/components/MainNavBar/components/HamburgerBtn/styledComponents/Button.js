/* eslint-disable prettier/prettier */
import styled from 'styled-components';
import CustomButton from 'commons/CustomButton/index';
import * as breakpoints from 'breakpoints';

const Button = styled(CustomButton)`
  display: none;

  .anticon {
    margin: 0;
    font-size: 18px;
    padding: 0;
    display: flex;
    align-items: center;
  }

  @media (max-width: ${breakpoints.xs}) {
    display: block;
  }
`;

export default Button;
