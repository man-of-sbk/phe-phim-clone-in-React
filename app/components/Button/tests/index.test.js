import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';

import Button from '../index';

afterEach(() => cleanup());

describe('<Button /> Component', () => {
  test('Should correct snapshot', () => {
    const { container } = render(<Button>text</Button>);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Should contain text chilren', () => {
    const { getByText } = render(<Button>button</Button>);
    const button = getByText('button');
    expect(button.textContent).toEqual('button');
  });

  test('Should correct text children when props children change', () => {
    const { getByText, rerender } = render(<Button>before change</Button>);
    getByText('before change');
    rerender(<Button>after change</Button>);
    getByText('after change');
  });

  test('Should test event click', () => {
    const mockFun = jest.fn();
    const { container, debug } = render(
      <Button onClick={mockFun}>button</Button>,
    );
    debug();
    fireEvent.click(container.firstChild);
    expect(mockFun).toHaveBeenCalled();
    expect(mockFun).toHaveBeenCalledTimes(1);
  });
});
