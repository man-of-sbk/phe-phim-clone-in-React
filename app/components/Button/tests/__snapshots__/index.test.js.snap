// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Component Should correct snapshot 1`] = `
<button
  class="ant-btn Wrapper-sc-1xlqh7-0 eeqAZY ant-btn-background-ghost"
  type="button"
>
  <span>
    text
  </span>
</button>
`;

